io/include/traccc/io/csv.hpp

```diff
@@ -288,7 +288,7 @@ inline cell_container_types::host read_cells(
 
             result.push_back(module, vecmem::vector<cell>({c}));
         } else {
-            auto idx = it - headers.begin();
+            auto idx = std::distance(headers.begin(), rit.base()) - 1;
```

Comment:
This is correct. But you should add just one line of explanation of why the `-1` is necessary. It's actually pretty educational to understand this. (Do you? :wink:)